File: app/src/main/java/datanapps/basicsample/ui/home/MoviesAdapter.java
package datanapps.basicsample.ui.home;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestBuilder;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

import datanapps.basicsample.R;
import datanapps.basicsample.utils.DNASnackBar;
import datanapps.basicsample.utils.Utils;

public class MoviesAdapter extends RecyclerView.Adapter<MoviesAdapter.MyViewHolder> {

    private List<Movie> moviesList;

    private Context context;

    public MoviesAdapter(Context context, List<Movie> moviesList) {
        this.moviesList = moviesList;
        this.context = context;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.layout_movie_list, parent, false);

        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        Movie movie = moviesList.get(position);
        holder.title.setText(movie.getTitle());
        holder.genre.setText(movie.getGenre());
        holder.year.setText(movie.getYear());
        Glide
                .with(context)
                .load(movie.getImgUrl())
                .apply(new RequestOptions().centerCrop())
                .into( holder.movieImage);
        holder.imgShare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Utils.shareAppDetail(context);
            }
        });
    }

    @Override
    public int getItemCount() {
        return moviesList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView title, year, genre;
        public ImageView movieImage;
        public ImageView imgShare;
        public MyViewHolder(View view) {
            super(view);
            movieImage = view.findViewById(R.id.movie_image);
            imgShare = view.findViewById(R.id.img_share);
            title = view.findViewById(R.id.title);
            genre = view.findViewById(R.id.genre);
            year =  view.findViewById(R.id.year);
        }
    }
}

File: app/src/main/java/datanapps/basicsample/ui/home/MoviesAdapter.kt
package datanapps.basicsample.ui.home

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import datanapps.basicsample.R
import datanapps.basicsample.utils.Utils

class MoviesAdapter(private val context: Context, private val moviesList: List<Movie>) : RecyclerView.Adapter<MoviesAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.layout_movie_list, parent, false)

        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val movie = moviesList[position]
        holder.title.text = movie.title
        holder.genre.text = movie.genre
        holder.year.text = movie.year
        Glide
                .with(context)
                .load(movie.imgUrl)
                .apply(RequestOptions().centerCrop())
                .into(holder.movieImage)
        holder.imgShare.setOnClickListener { Utils.shareAppDetail(context) }
    }

    override fun getItemCount(): Int {
        return moviesList.size
    }

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var title: TextView
        var year: TextView
        var genre: TextView
        var movieImage: ImageView
        var imgShare: ImageView

        init {
            movieImage = view.findViewById(R.id.movie_image)
            imgShare = view.findViewById(R.id.img_share)
            title = view.findViewById(R.id.title)
            genre = view.findViewById(R.id.genre)
            year = view.findViewById(R.id.year)
        }
    }
}